### DISCUSSION TITLE
能否请教一下作者大大，怎么修改JS文件，让通过IDM下载的EF2文件中包含filename: number-title.format

### DISCUSSION THREAD

**Discussion by LibertyCola** [the1812/Bilibili-Evolved#2871] [提问 / Q&A] [answered]
找到一个ef2协议工具，可以让IDM批量下载时标题名称正确，但是需要在生成的ef2文件中增加 filename: number-title.format，苦于JS不太懂，希望大佬可以指教一下，谢谢。

修改后格式如下
<
https://xy115x56x91x4xy.mcdn.bilivideo.cn
referer: https://www.bilibili.com/bangumi/play/ep327
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0
filename: number-title.format
>

**Comment by the1812** [comment, marked as answer]
我来加吧, 你可以测试下加了之后的插件: [idm-output.zip](https://github.com/the1812/Bilibili-Evolved/files/7835142/idm-output.zip)

**Reply by LibertyCola** [reply]
谢谢大佬，基本能够正常工作了

**Reply by LibertyCola** [reply]
非常感谢，完美适用/赞

**Reply by xlsufo** [reply]
> 非常感谢，完美适用/赞
请问你使用的ef2协议工具是什么呀，我也想尝试一下

**Comment by wangfeng35** [comment]
奇怪 我试了 怎么文件名没变

**Reply by xlsufo** [reply]
我也是，文件名没有按照filename改变

**Comment by Young-Lord** [comment]
必须结合此工具使用，否则是无效的： https://github.com/MotooriKashin/ef2
